### PR TITLE
Api - UnitTest - Search Group and Remove user from group

### PR DESCRIPTION
**Pull request by at-hoavo**
#### Related issues (Required)
- Task #225 #244
- Bug #

#### Description (Required)
- [x] Create test  api

### CODE REVIEW

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information. (2/22)</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please rebase to get rid of the merge commits in this PR</td>
    </tr>
  </tbody>
</table>

Test case | Classname|
--- | --- |
Given a group name - When call trigger search group - Then return list Group | vn.asiantech.way.group.search.SearchGroupViewModelTest|
Given mock response - When search group - Then return SearchGroupResult object | vn.asiantech.way.api.HypertrackApiTest|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/main/java/vn/asiantech/way/data/model/Group.kt [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
+import com.google.gson.annotations.SerializedName
 import java.io.Serializable
 
 /**
  * Copyright © 2017 Asian Tech Co., Ltd.
  * Created by cuongcaov on 09/11/2017
  */
 data class Group(val id: String, var name: String, var token: String,
-                 var ownerId: String, val createAt: String, var modifiedAt: String) : Serializable
+                 var ownerId: String, @SerializedName("created_at") val createAt: String,@SerializedName("modified_at") var modifiedAt: String) : Serializable
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/model/Group.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/e40b991104c20bad9bad4c24befb74c55dbb2f1c/app/src/main/java/vn/asiantech/way/data/model/Group.kt#L11"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/search/SearchGroupFragmentUI.kt [hunk 2]
```diff
@@ -9,13 +10,16 @@ import org.jetbrains.anko.*
 import org.jetbrains.anko.recyclerview.v7.recyclerView
 import org.jetbrains.anko.sdk25.coroutines.onClick
 import vn.asiantech.way.R
+import vn.asiantech.way.data.model.Group
+import vn.asiantech.way.data.model.Invite
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/search/SearchGroupFragmentUI.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/81d82881bac091c54bdf7697fc8a43d7a71d665c/app/src/main/java/vn/asiantech/way/ui/group/search/SearchGroupFragmentUI.kt#L14"></span>Unused import</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/ui/group/search/SearchGroupViewModel.kt [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package vn.asiantech.way.ui.group.search
 
+import android.support.v7.util.DiffUtil
+import android.util.Log.d
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/ui/group/search/SearchGroupViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/81d82881bac091c54bdf7697fc8a43d7a71d665c/app/src/main/java/vn/asiantech/way/ui/group/search/SearchGroupViewModel.kt#L4"></span>Unused import</td>
    </tr>
  </tbody>
</table>

**Comment by at-hoavo** [comment]
@at-vinhhuynh please review for me, thanks very much!

File: app/src/main/java/vn/asiantech/way/data/source/WayRepository.kt [hunk 4]
```diff
@@ -50,15 +52,23 @@ class WayRepository : WayDataSource {
...
     }
+
     override fun createAndAssignAction(builder: ActionParamsBuilder): Observable<Action>
             = remoteDataSource.createAndAssignAction(builder)
+
+    override fun getTrackingURL(): Single<String> = remoteDataSource.getTrackingURL()
+
+    override fun getLocationName(context: Context, latLng: LatLng): Single<String> = remoteDataSource.getLocationName(context, latLng)
+
+    override fun getCurrentHyperTrackLocation(): Single<HyperTrackLocation> =
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/WayRepository.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/8a51222870fc25d58e3c4a1b7af4fa9c1241d379/app/src/main/java/vn/asiantech/way/data/source/WayRepository.kt#L69"></span></td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt [hunk 5]
```diff
@@ -131,12 +137,13 @@ internal class WayRemoteDataSource : WayDataSource {
...
-    override fun getCurrentLocation(): Observable<HyperTrackLocation> {
-        val result = AsyncSubject.create<HyperTrackLocation>()
-        HyperTrack.getCurrentLocation(object : HyperTrackCallback() {
+    override fun getETA(destination: LatLng, vehicle: VehicleType): Observable<Float> {
+        val result = AsyncSubject.create<Float>()
+        HyperTrack.getETA(destination, vehicle, object : HyperTrackCallback() {
             override fun onSuccess(response: SuccessResponse) {
-                val res = HyperTrackLocation((response.responseObject) as? Location?)
-                result.onNext(res)
+                response.responseObject.let {
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/8a51222870fc25d58e3c4a1b7af4fa9c1241d379/app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt#L144"></span>Cast operator throws an exception if the cast is not possible.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt [hunk 6]
```diff
@@ -165,20 +172,67 @@ internal class WayRemoteDataSource : WayDataSource {
...
+        return locationName
+    }
+
+    override fun getCurrentHyperTrackLocation(): Single<HyperTrackLocation> {
+        val result = SingleSubject.create<HyperTrackLocation>()
+        HyperTrack.getCurrentLocation(object : HyperTrackCallback() {
+            override fun onSuccess(p0: SuccessResponse) {
+                try {
+                    p0.responseObject.let {
+                        HyperTrackLocation(it as Location?)
```
Comment (at-dev-support-01) [inline app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/akaj-team/project-way/blob/8a51222870fc25d58e3c4a1b7af4fa9c1241d379/app/src/main/java/vn/asiantech/way/data/source/remote/WayRemoteDataSource.kt#L213"></span>Cast operator throws an exception if the cast is not possible.</td>
    </tr>
  </tbody>
</table>

**Pull request closed without merging**